### PR TITLE
Add auth0 integration

### PR DESCRIPTION
**Pull request by Aumarh**
For your review

### CODE REVIEW

**Review by hynnot — COMMENTED** [review]
We'll review this together tomorrow because there are some things I don't understand.

File: backend/src/app.ts [hunk 1]
```diff
@@ -1,46 +1,76 @@
...
-        console.log('Failed to do something async with an unspecified error: ', error);
-        return res.send(500);
-    }
+const checkJwt = auth({
+  issuerBaseURL: 'https://localhost:5000',
+  audience: 'butterfy',
+});
+
+// This route doesn't need authentication
+app.get('/public', function (req, res) {
```
Comment (hynnot) [inline backend/src/app.ts:29]: We can remove this because we have other routes publics

File: backend/src/app.ts [hunk 2]
```diff
@@ -1,46 +1,76 @@
...
\ No newline at end of file
+// This route needs authentication
+app.get('/private', checkJwt, function (req, res) {
+  res.json({
+    message:
+      'Hello from a private endpoint! You need to be authenticated to see this.',
+  });
+});
+
+app.listen(3000, () => console.log('listening at http://localhost:3000'));
```
Comment (hynnot) [inline backend/src/app.ts:44]: Remove this, we have this in server file.

File: backend/src/auth/auth.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const express = require('express');
+const app = express();
+
+const port = process.env.PORT || 8080;
+
+const jwtCheck = jwt({
```
Comment (hynnot) [inline backend/src/auth/auth.js:6]: I think we need to move this seetings to the app file.

File: backend/src/server.ts [hunk 4]
```diff
@@ -1,20 +1,20 @@
-import app from "./app";
 import dotenv from 'dotenv';
-import { AppDataSource } from "./data-source"
+import jwksClient from 'jwks-rsa';
```
Comment (hynnot) [inline backend/src/server.ts:2]: I think we don't need this here

**Review by hynnot — CHANGES_REQUESTED** [review]

File: backend/src/app.ts (hunk 1 above)
Comment (Aumarh) [reply backend/src/app.ts:29]: done

**Review by hynnot — APPROVED** [review]

**Pull request merged**